7.7

ns1045 opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/8720)
<!-- Reviewable:end -->

Flamenco committed:
Add method to auto scale calendar time range (#17715)

Backported from master branch (Vaadin 8).

Change-Id: I363c9798de0d238a5a4aa4b4e839a31da460434f

emarc committed:
Add valo-menu-hover stylename to make responsive menu appear w/o adding code (#19019)

Adding the optional "valo-menu-hover" style to the "valo-menu" element
will make the menu appear on hover - i.e when mousing over on desktop,
when tapping on touch devices (which also triggers hover).

Backported from master branch (vaadin 8).

Change-Id: I8344234a20aa93216b0116608d81e30e86d8b309

mstahv committed:
Fire actions before removing menu from the DOM (#20080)

Change-Id: Ic1cd285c697a05d186d81935f632881e4cf561bf

thevaadinman committed:
Discard for DateField when the data source contains null (#8069)

Earlier discard did not work (#8069) if the data source
contained null and the user entered an invalid value for date fields.
The discard logic has been changed such that the error indicators and
internal state of the date field is cleared also if the value entered by
the user is invalid and the data source is null. 

attach() updates the value from the data source in the same way as
discard. The old logic in attach is retained. 

Change-Id: I5d798a2fdc3d2f0e82a037ea3cb2f57b6432039e

Artur- committed:
Prevent Atmosphere from resurrecting a destroyed broadcaster (#20026)

Change-Id: Ifdf6d8f61eb1f5f19736e057e9314b400d6030ee

Commit:
Start client-side Grid DOM event handling refactor

Change-Id: I67dad4cfbd6e9e355c48703253ff3d6da00e4eef

Commit:
Refactor Grid event handling methods to instances of a handler interface

Change-Id: Icbed14cbef14c434f1909fa0350cb0364235c774

wbadam committed:
Fix Grid height calculation when HeightMode.ROW (#20104, #20074)

When Grid is inside of a Tab (Tabsheet or Accordion) and
height mode is set to HeightMode.ROW, layout happens
before row height is known. To fix that, an event is fired
and a handler starts a layout after everything is calculated.

Change-Id: Idc8de795956b387ec69adf1871cb7f557914d998

Commit:
Update expected Chrome version to 52

Change-Id: Ic580c17bddf05455827ecb7fbfb44e5c9e2c7bf2

Commit:
Remove unused GWT dependencies

As vaadin-client-compiler dependens on gwt-dev, the gwt-dev dependencies
are either bundled in gwt-dev or specified as transitive dependencies
for it, so there is no need to specify them again for
vaadin-client-compiler

Change-Id: Iec075a3b4a005ae7b2e331287d942357aafe277d

Artur- committed:
Fix OSGi manifest for vaadin-shared

The shared package seems to no longer use any
flute or streamhtmlparser classes

Change-Id: Ie10f0b8169ec79afacc5bf1dc7a30c7eb809a413

Commit:
Update release notes for 7.7.0.rc1

Update release type, ticket list link and supported browsers.

Change-Id: I4e05620e78f081c70c0efc1545f7a6a41d090ba6

Commit:
Update/remove empty @since for 7.7

Change-Id: Ib46a450c7d6a6d15eb6bea5d53351d182929e52f
(cherry picked from commit b322664b8b21da53e6fc833005fe3968f8352bec)

Artur- committed:
Remove confusing note about OSGi support (#20094)

Change-Id: I8c20a2e3ac13fddf1ff207ae8da88ad822e5787c

mstahv committed:
Added a fallback value for useragent (#20172)

Without this change, some browsers, like GoogleBot, don't properly render the page.
Instead of weird JS exception (and hit to server with 404) we should at least try
to render the app with all browser. As webikit/chromium based browsers are
nowadays most popular, falling back to "safari" permutation.

Change-Id: I4275110962432b2552d51715a86a8e815fe539ad

Commit:
Add release notes mention about strict mode

Change-Id: I93eec041fbccd3961e82f2d7e52d23eb044ec4ea

Artur- committed:
Properly encode special UTF-8 characters (#19928)

The main encoding problem was fixed in GWT 2.7.0.vaadin4.
Vaadin 8 is already fixed as it uses GWT 2.8

Change-Id: I78ea30c1ff81d44432cd82ecd23eda8eb86e6efb

Commit:
Update release notes for maintenance releases

Change-Id: I755dfe84224f32f0102661aae228656174fb7c76

Artur- committed:
Require correct servlet version in OSGi (#20190)

Change-Id: I87c2acd418f52349fba6fd11eba0530cdb5ef5c3

Artur- committed:
Use Vaadin plugin 7.7.0

Change-Id: Ia7d1671670c9c6e06328612d980ea2d455eaa0d1

Commit:
Updated tutorial to Vaadin 7.7.0

Change-Id: I1ef5a89a7e9f700a9302ff535d7e30b692316e22
(cherry picked from commit 09909cdcc47479c6b6f3ecb633ed4128f40f7b3c)

Artur- committed:
Fix animation end listeners so they are always removed (#17903)

Fixes ComboBox suggestion popup so that it will not automatically
close when clicking the popup button, if the user happened to
double click on the button earlier.

Change-Id: I60936f82133ca89c2aef0dccff68092a11deff1c

Artur- committed:
Use servlet context classloader when finding servlet class for websockets (#20024)

Change-Id: Ia885ecebe854c0aeda47be2bbc9a9d8a47a03d5c

Artur- committed:
Do not run test on IE8 as IE8 is broken

Change-Id: I55f41c8f3488a57e6ab499da58df70eb02d3f136

Artur- committed:
Fix ComboBox paging when number of items equals page length (#20213)

When pageLength == number of items, no paging should be shown except
if null selection is allowed, then there should be a second page
with one item.

Change-Id: I01c00f0c9d75a1dbb80d11b07c37c8ad7334ea07

Artur- committed:
Update DOM and update escalator row count in the correct order (#19442)

If you show a column when Grid has no horizontal scrollbar, and is scrolled
all the way down, and the number of visible rows is slightly more than N then
Escalator will adjust the row count when the scrollbar is shown so that N-1
rows are visible. During this operation, the DOM must be updated for the new
column.

Change-Id: I0b6c845f96a57be1d64ef4e735aa2f77efbe589a

Artur- committed:
Prevent editor from being canceled while it is being saved (#19458)

Change-Id: I062e097134035856f6c120584f48a5f3601bd9ac

Artur- committed:
Delete broken stylesheet and revert to default style until a new stylesheet is created

Change-Id: I8a2a3e2e9c35c38f58e0a64fa5e0363702e54e0f

Artur- committed:
Always show loading indicator for JavaScript RPC (#20235)

The change to not show loading indicator for JS RPC was done in
15a53d305e6648572c11cb988f4988c2c5825af1 and looks like a refactoring
error.

Change-Id: If2a2818e0c4adc06479ecaf17764bf53fc89c095

Artur- committed:
Ensure Firefox always updates the grid scrollbar (#19802)

When the scrollbar is scrolled a bit down and the number of items
in the grid is reduced so that no scrollbar is needed anymore, it seems
that Firefox refuses to send a scroll event even though the scroll position
is updated before the scrollbar is hidden.

Change-Id: I626536a9efd036bc826b1e6be3363332a56774f6

Artur- committed:
Calculate column widths immediately if there is data (#20269)

Deferring the column width calculations, like done before,
causes flickering as each column is first rendered with its
minimum required width.

Change-Id: I02e6951a99dcb0cb9ad78a95915076d064fa9642

Artur- committed:
Test for Firefox download disconnecting push channel

Change-Id: Iacf2b04b96c237ad377dbd75ad07f86a00660319

Artur- committed:
Do not include yuicompressor for Sass compiler

The yuicompressor package is needed for optimizing the CSS output,
something which is not needed for on-the-fly compilation in
development mode.

Removing yuicompressor and its dependencies reduces the size of all
WAR files by 1.6MB

Change-Id: Ie5058ae9c22f580d358e2cd7d9d34a8f39b1feac

Artur- committed:
Make test independent of any converters present in the factory

Change-Id: I8b6b45df9d3929e4447976b8c7ae72e6491fb1af

Artur- committed:
Change snapshot version 7.7-SNAPSHOT

Change-Id: I7929f6fbe9a83f63fcb7c1f3b174dd84d4add185

Artur- committed:
Handle encoded URL characters correctly when constructing widget set name (#20262)

Change-Id: Idfdd5e6021444af5b08d48ddf90745b47ff4643d

Artur- committed:
Skip Maven enforcer plugin during demo validation

Change-Id: Ic4b7dc9f9b646d88fe6b323d2c051f86edeeaa70

Artur- committed:
Make test pass on all browsers

Change-Id: I8226ce8983eb72e0afb4948fc644cb587bc01708

Artur- committed:
Build demos from 7.7 branch

Change-Id: Ic2a0396ad451adfd2c75b3dbd54155d28fb16be7

Artur- committed:
Use widget set specified by init parameter (#20276)

1. If a UI has @Widgetset, use that
2. If a "widgetset" init parameter has been specified, use that
3. If there is an AppWidgetset, use that
4. Use DefaultWidgetSet

This changes the default value of the @VaadinServletConfiguration widgetset
parameter to "", to avoid always setting the "widgetset" init parameter when
@VaadinServletConfiguration is used.

Change-Id: I87f9aa294d31e62bb94b7e831dd7b25ca1ce483b

Artur- committed:
Use widget set specified by init parameter (#20276)

1. If a UI has @Widgetset, use that
2. If a "widgetset" init parameter has been specified, use that
3. If there is an AppWidgetset, use that
4. Use DefaultWidgetSet

Change-Id: I522210220bb9e3eb339e9d226833d0fa8eb65839

Artur- committed:
Use correct column index when calculating min width during resize (#19833)

Change-Id: If7a7a5c1e5eff4bdac0f6e5943a9fac2ce98784a

ingokegel committed:
Fix regression that broke widget set compilation in 7.7.1 (#20285)

The fix for

https://dev.vaadin.com/ticket/20262

broke widget set compilation on Windows, possibly only under certain circumstances

Change-Id: Ica559b9c7199014bf1a2babad29f53b083ff74a2

Artur- committed:
Ensure temporary layout manager state is cleared at the end of a layout phase

Change-Id: I28f3d20e6db999a033429338535719f14e495d4f

Artur- committed:
Fix assertion error when column widths are calculated

Change-Id: I042a5bfb548b52911100e01e5a74901a1abfbf33

Artur- committed:
Make initially disabled grid work when enabled (#20240)

Adds API to allow grid to ask its data source if it is waiting for data.
The previous tracking inside grid was not always correct as it relied
on dataAvailable being called.

Change-Id: I0bdb448d9b720155940b0834e118f0eca27a3bfc

Artur- committed:
Format Java files using Eclipse Neon and Vaadin settings

Using the same formatting in the 7.7 branch and the master branch
eases porting of fixes

Change-Id: I51b4693f2f120447702cf322ce4505608ad92987

Artur- committed:
Change javadoc to a style Eclipse formatter can handle

Change-Id: I2ad135c72821b6e194fd77bd141b638c36e138a7

Artur- committed:
Use requestAnimationFrame when scrolling in Grid (#20294)

Chrome no longer (since version 5x) always fire deferred commands
immediately during scrolling but can delay them with several hundred
milliseconds, making grid really unresponsive.

So far, nobody has been able to provide a reliable test case.

Change-Id: Ide80aef2d661c9e27b67c8e62e85734af7a38cab

Artur- committed:
Specify branch also for Sampler

Although sampler is pre cloned, the script still tries to
mess with the defined branch (master is not found in the cloned
repo)

Change-Id: I91f31fe0a276221cc357cf0678759ad4125c518e

Artur- committed:
Update to Chrome 53

Change-Id: I518b3f261adb633bb09a2597c9e7ee3cb6d63e6a

Commit:
Updated link to new step 1 video in tutorial (the new video uses Vaadin 7.7 series)

Change-Id: Ic40b70e692e686d638e163a224816aabeb6de100
(cherry picked from commit e92a737fe8b0655bbf0c3ce1881777d418147b41)

Legioth committed:
Revert "Use widget set specified by init parameter (#20276)"

Breaks all use of AppWidgetset together with
@VaadinServletConfiguration

This reverts commit 516c428ca127e3c31b7b4d74220e4b7eed4571be.

Change-Id: I9e9bd792479ee1f4dde08c5d2a4d87e06b77f519

Artur- committed:
Fix widget set builder to create widget set in correct location (#20320)

The fix for #20285 generates URLs so that the end in a slash, where the old
version generated URLs without the trailing slash. The trailing slash was not
taken into account by the code which updates a widget set.

Change-Id: Iabe716ed1632da7b2a6b1f4ffdc987c745d9452d

Artur- committed:
Format project pom files using correct settings (4 spaces)

Change-Id: Ib8ae22a56334046effe721bb4d3769ef7b92d61d

Artur- committed:
Add tests for widgetset compilation in different modes

* Using the default widget set
* Using no add-ons but a custom widget set
* Using add-ons with a defined widget set
* Using add-ons with no defined widget set
* Using directories with special characters (space)

Change-Id: I85450de359f410003bed94b120915b4736bc2d4f

Artur- committed:
Use versions-maven-plugin 2.3 to avoid NPE while setting project version

Change-Id: I9080a9f46193c1dea4a9ef6856d2899fe960b7e9
(cherry picked from commit e7b9db54a5aa172509b33746d270ac375009b4a6)

Commit:
Fixes VTree so that you can click sub elements when setHtmlContentAllowed is true. (#20097)

Change-Id: Ic453aaf16ea1d44792db868e04faaba069aab3b7

johannest committed:
Fixed touch scrolling issue in Surface and WP devices (#18737)

Fixed by using mouse events instead of touchevents when the browser is
IE or Edge (added isIEOrEdge utility method in the BrowserInfo). Also
added touch-action: none; css rules into escalator.css to prevent
default touch behaviour on IE and Edge
No new unit tests since we do not have automatic touch testing
possibilities yet.
Please test manually with Surface: IE and Edge, use for example uitest:
com.vaadin.tests.components.grid.basics.GridBasics

Change-Id: I5b37f1648e2051ea0ab4d56ab767186b532da07c

Artur- committed:
Allow defining a focus delegate component for CustomField (#20336)

Change-Id: I1160e7a384b1816204eb7f4b0f52f83ed9e230c0

Artur- committed:
Workaround for deadlock issue (#18436)

Change-Id: I4e32550e3d3095c2c914bb93d260819414d2e6e6

Artur- committed:
Make vaadin-server export the VAADIN package again (#20332)

Change-Id: I7241ec11a9bfe7ebfa6a720b1abb3c4d0fabdb36

Artur- committed:
Add Grid.refreshRows to allow refreshing individual rows (#16765)

Change-Id: I554d7b6ca60840bab3a032daa4847e7271086e27

Artur- committed:
Make modal window block shortcuts for underlying components (#20366)

Change-Id: Ibb3b52b652d77a1bfa786417b1c92fd0a578035f

Artur- committed:
Make clicking outside a ComboBox add a pending new item (#18366)

It should not matter how you move from a field to the next. Tabbing
to the next field already previously caused the new item handler to
be invoked. This fixes the case when you click in the next field to
work the same way.

Change-Id: I3978d87cde8f05dce92edda2a4a116f4a15e0749

Artur- committed:
Check for duplicate property ids when setting Grid columns or column order (#20386)

Change-Id: I76be83642f0e56e55b0c0e502ac6769de1ee8af0

Artur- committed:
Make UPPER_CASE_PROPERTY_IDS more human friendly by default (#20380)

Change-Id: I9bd845a08fa3c440f05d102ccf8920dfc1c95b6b

Artur- committed:
Never check the selectAll checkbox in an empty grid (#20301)

Change-Id: Ia15edb5a10de9bff047044161ffe83dd09f51098

Artur- committed:
Keep expand ratio for last row/column when reducing grid layout size (#20297)

Change-Id: Iff53a803596f4fc1eae8e4bfa307b9c1f4df961a

Artur- committed:
Allow a resize listener to fire a resize listener (#20338)

Change-Id: I6b045c3a693e88b3cba182ae9ea68cfa98c914b0

Artur- committed:
Send selection change events when changing selection mode (#20391)

Change-Id: Iaeb772981f3ac039bf0fe3cde595cfdb3691cddf

Artur- committed:
Update client side selection state when changing selection model (#20370)

Change-Id: I7fe8c160405a69140cac511c744d89a62c401cca

wbadam committed:
Fix NPE in case some items don't contain all properties of Grid.
This could occur in when parent is a different entity than its children
in hierarchical data.

Change-Id: Icd53b5b5e5544a3680d0cd99702ab78224b2dc08

Artur- committed:
Make Grid isWorkPending wait for scroll events to complete (#20417)

Change-Id: I3cfdac539111ff2e88c7ff81e6d6b9a579160423

Artur- committed:
Publish Escalator.getBodyRowCount to JS (#20344)

This is needed to be able to get the information from e.g. TestBench

Change-Id: I95ec6064f602e6a8b24e35cc59d467e9c8b31c1e

Commit:
Update Hierarchical container JavaDocs to match code (#5864)

Change-Id: I7a8520ef61ddc4dff66c326d61acbf7ccf0e76cd

Artur- committed:
Ensure #cell[N] always scrolls row N into view (#20423)

Change-Id: I99347ee1d0f2f13fcb8e110d2e7192590e4f9456

Artur- committed:
Use US english (license) in all places

Change-Id: I5fa99627a1c3302dd905d78149e5f32c7315a5c7

Artur- committed:
Mention latest browser versions

Change-Id: I06477d7ee5252ec4a5e7fdbc05fe6e31e05782e6

Commit:
Date/time issues are fixed in the tests.

- legacy DateField declarative format tests are fixed (day light saving
time is taken into account).
- CalendarMonthViewDndEvent UI because you are not supposed to work on
Sundays (when the test is broken).

Backported from master branch (Vaadin 8).

Change-Id: I5e62c09da579e9f6eb2a58a87253618d3f30b2e9

Artur- committed:
Show a sensible message for missing extensions (#10799)

Also remove ComponentMissingFromDefaultWidgetsetTest since it's
identical to UnknownComponentConnectorTest

Change-Id: I4b4b8f40c8376f4ea26b73d41191a3e7e811df01

thevaadinman committed:
Add lazy/simple resize mode to Grid (#20108)

Change-Id: I47427efc28c350382dba8c1f50fd332a3f4585e4

Commit:
Prevent adding several scrollbar handlers (#19189).

Change-Id: Ib0cc6c6835aab6d263f153362a328bcf2be7bc5c

Commit:
Correct addon init param widgetset IT test.

Change-Id: I413481a325ebad15aa16dd639c788cc72f18dc52

Commit:
Fix modern browsers test failures for 7.7.

Change-Id: Ibef57aa4f8838f3bbc5f5c5500f3f4f7fc6eb438

Commit:
Update @since tag for 7.7.4.

Change-Id: Ic6711221d0f09ad369c4074d805b237eccb2e5a7

Commit:
Fix some modern browsers tests failures.

Change-Id: I2a89d3493e2f98ab1b9a0666d9dc075a2457e432

Commit:
Update Vaadin Spring documentation for 1.1.0

Change-Id: I8dcf5b7576a64b26419136ca6b8927f4c2731d60

Commit:
Initial versions of the new release report scripts

These have been backported from Vaadin 8 branch with minimal
modifications.

Change-Id: I292917aa0457c69b569313a3e56913b4a8d85316

Artur- committed:
Revert "Fixed touch scrolling issue in Surface and WP devices (#18737)"

The patch prevents Escalator and Grid from ever being rendered on IE8 because element.addEventListener does not exist there

This reverts commit eb2f5ac3a714aefce948d447a4004b94e35e44a9.

Change-Id: I58f6db7432a1cd10299891b2d8c55c855e87265f

Commit:
Fix python list syntax in BuildDemos

Change-Id: I42f2f3fa3597f67f259c3ff9f158560303ffb328

Commit:
Fix archetype list in build report script

Change-Id: Ie215346122ad6533176fd2a8d71660805eacb942

wbadam committed:
Fixed drag and drop failure when message dragged from email client (#20451)

When dragging message form email client on Windows, item.webkitGetAsEntry()
might return null creating NPE on the client side. Added additional checks
for this situation.

Change-Id: I569f7e6d0d7b137f24be53d1fbce384695ae8c73

Commit:
Fix links in build report

Change-Id: Id32467982207ac9a9ce45301097652715d9dfff8

johannest committed:
Fixed touch scrolling issue in Surface and WP devices (#18737)

Fixed by using pointerevents instead of touchevents when the browser is IE11, 
or Edge. Also added touch-action: none; css rules into escalator.css to prevent 
default touch behaviour on IE11 and Edge. Does not affect IE8 to IE10 browsers,
behaviour on those will stay the same as before the fix.

No new unit tests since we do not have automatic touch testing possibilities yet.
Please test manually with Surface: IE11 and Edge, use for example uitest:
com.vaadin.tests.components.grid.basics.GridBasicsomponents.grid.basics.GridBasics

Change-Id: Iddbf1852e6ffafc855f749d6f4ebb235ed0f5703

Commit:
Change expected pre-release version number pattern in publish report

Change-Id: Icdacecc490d2490ea9e262f5c5736c1dece2a89d

Artur- committed:
Mark TextField/TextArea as busy when a text change event is pending (#20469)

Change-Id: I404985ae0be1e7dc65171b610032f8649e700f50

Commit:
Remove CustomFieldState (#20468)

Moves FocusDelegate to AbstractFieldState.
This change is done to revert backwards compatibiity breaking API
change in 7.7.4 when CustomFieldState was introduced.

Change-Id: I76212a8700463140520d5482b740fe0edbfb082c

Artur- committed:
Re-add body rows when Escalator is reattached to DOM (#20477)

Change-Id: I0ae9144817db3bb730c80748d5e9190484b323e7

Commit:
Update @since tags for 7.7.5

Change-Id: I73201908b537ca0a65106e03c9bad913b59fae9f

Commit:
Fix wrong year in copyright header

Change-Id: Idbf6c1f978b9b36ed82bf8b96a6a649ae4772ae6

tsuoanttila committed:
Add screenshots as a Git submodule (#86)

ahie committed:
Move Grid event handlers to a list, add method to add custom handlers (#76)

* Move Grid event handlers to a list, add method to add custom handlers

Change-Id: I2fe67ae961bfb0f7c44ba704346762a0faf6da2c

* Merge branch '7.7' into grid-event-refactoring

tsuoanttila committed:
Fix java packaging order (#106)

Closes vaadin/archetypes#113

Commit:
Fix margin in declarative format is recognized as unsupported property gridlayout (#103) (#7952)

Backported to 7.7 from master (8).

Fixes vaadin/framework8-issues#465

Change-Id: I9734341b747b8d16020443c71d72fcfd7e331af7

Legioth committed:
Use proper UTF-8 encoding for Content-Disposition filenames (#19527) (#6607)

Artur- committed:
Enable changing the backing bean for BeanItem (#4302) (#77)

When storing a bean to the database, you typically get a new and updated
bean instance back. By allowing to change the bean instance, we make it
possible to just update the single BeanItem instance which can be used
in many places.

Artur- committed:
Replace non-breaking spaces with space (#7957)

Artur- committed:
Make AtmospherePushConnection methods public (#7973)

There is no sensible way to use a custom version of APC, so protected
access does not help in any way to access the underlying resource and/or
connected UI.

Commit:
Fix context menu missing box-shadow for Runo on Chrome (#7966)

Change-Id: I318761527a695c741f52b4c18e9f8ea943f89017

Artur- committed:
Use correct indexes in multiselect checkboxes after removing rows (#8072)

Fixes #8011

Artur- committed:
Fix removal of hidden Grid columns (#8071)

Fixes #8018

Commit:
Fix chrome version (#8086)

* Fix chrome version

Artur- committed:
Call error handler for exceptions in UI.init() (#8055)

Fixes #4995

Artur- committed:
Render font icon correctly on the 'more' menu item (#8126)

* Render font icon correctly on the 'more' menu item

Fixes #8125

OlliTietavainenVaadin committed:
reopen Grid details on attach, fixes #8015 (#8074)

Fixes #8015

Artur- committed:
Correctly detach components in merged cells when a static row is removed (#8142)

Fixes #8140

Artur- committed:
Fix exception when Grid is inside a PopupView (#8157)

Fixes #7695

ahie committed:
Remove tracking of unregistered connectors (#8153)

* Remove tracking of unregistered connectors

* Merge branch '7.7' into 8111-remove-unregistered-connector-tracking

* Merge branch '7.7' into 8111-remove-unregistered-connector-tracking

* Add tests that verify markAsDirty is called on old parent

* Merge branch '7.7' into 8111-remove-unregistered-connector-tracking

Artur- committed:
Reset selected icon when blurring a ComboBox (#8082)

* Reset selected icon when blurring a ComboBox

Fixes #8073

* Merge branch '7.7' into _combobox-icon-reset-selection

* Merge branch '7.7' into _combobox-icon-reset-selection

Artur- committed:
Make it possible to deselect what was selected on the server (#8235)

* Make it possible to deselect what was selected on the server

Fixes #8221

Artur- committed:
Properly report invalid arguments to StaticRow.join() (#8243)

* Properly report invalid arguments to StaticRow.join()

Closes #8234

alejandro-du committed:
Removed broken/obsolete 'Setting up the Development Environment' link (#8251)

* Removed broken/obsolete 'Setting up the Development Environment' link

Artur- committed:
Properly end HTML5 drag operations when no drop is performed (#8242)

* Properly end HTML5 drag operations when no drop is performed

No auto test because of https://github.com/SeleniumHQ/selenium/issues/1365

Fixes #4060

Artur- committed:
Rename variables to indicate whether they use DOM index or Grid index (#8275)

ahie committed:
Fix closing of modal window curtains while dragging and resizing (#8281)

Fixes #7496

Artur- committed:
Fix navigation using keyboard to frozen column when there are hidden columns (#8277)

Artur- committed:
Flush the correct editor widget if there are hidden columns (#8276)

Fixes #8266

Artur- committed:
Serve VAADIN files also from META-INF/resources (#8286)

Fixes #8206

caalador committed:
Fix absolute layout component sizing on size change in Vaadin7 (#8304)

Changing from relative to defined size leaves the component positioned wrong,
also changing from defined size to relative size leaves the component in the
wrong position with the wrong size.

Change makes the wrapper rerun setChildWidgetPosition() that adds/removes
necessary css definitions to/from the cssPosition String.

Artur- committed:
Remove invalid trailing comma from Thai date format (#8311)

* Remove invalid trailing comma from Thai date format

Fixes #8303

wbadam committed:
Improve Grid extendability (#8343)

* Improve Grid extendability (#8342).
Find GridState field type using getField() instead of getDeclaredField() so that the field could be found in superclass as well.

Change-Id: I33996f45e848575d11bb206c6a1d30d19e514968

Artur- committed:
Make it possible to disallow user selection in Grid (#8144)

Fixes #7880

Artur- committed:
Prevent reorder of frozen columns when there are hidden columns (#8278)

* Rename variables to indicate whether they use DOM index or Grid index

Grid indexes include hidden columns, DOM/Escalator indexes do not

* Prevent reorder of frozen columns when there are hidden columns

Fixes #8265

* Merge branch '7.7' into _grid-no-reorder-of-frozen-columns

* Merge branch '7.7' into _grid-no-reorder-of-frozen-columns

* Merge branch '7.7' into _grid-no-reorder-of-frozen-columns

* Merge branch '7.7' into _grid-no-reorder-of-frozen-columns

* Merge branch '7.7' into _grid-no-reorder-of-frozen-columns

* Merge branch '7.7' into _grid-no-reorder-of-frozen-columns

* Merge branch '7.7' into _grid-no-reorder-of-frozen-columns

Artur- committed:
Add method for refreshing all Grid rows (#8406)

* Add method for refreshing all Grid rows

Fixes #8350

Artur- committed:
Use an actually transparent 1x1 image (#8407)

Fixes #8348

Artur- committed:
Call "destroy" instead of "destory" in JavaScriptRenders (#8263)

* Call "destroy" instead of "destory" in JavaScriptRenders

For backwards compatibility, calls "destory" if no "destroy" exists

Fixes #8162

* Merge branch '7.7' into _js-renderer-destory

* Merge branch '7.7' into _js-renderer-destory

* Merge branch '7.7' into _js-renderer-destory

* Merge branch '7.7' into _js-renderer-destory

* Merge branch '7.7' into _js-renderer-destory

Artur- committed:
Fix caption updating in CustomLayout (#8415)

Fixes #8413

caalador committed:
Fix the absolute layout component resize positioning tests. (#8456)

* Fix gthe absolute layout component resize positioning tests.

caalador committed:
Fix problem with re-opening the popup (#8446) (#8454)

Fixed problem that disables opening of popup
after closing it by clicking the datefield-button.

Commit:
Update 7.7 release notes to github era (#8472)

* Update 7.7 release notes to github era

Fixes framework8-issues#222

Commit:
Update missing since tags for 7.7.7 (#8474)

* Add missing since tags for 7.7.7

caalador committed:
Fix toggling of WeekNumbers for PopupDateField. (#8405) (#8469)

Refresh calendar body for show week numbers state change.

caalador committed:
Fix absoluteLayoutResizeComponents test for IE8 (#8497)

IE8 handles the sizings differently from other tested browsers so it
needs to be tested on 'left: auto' and not wrapper div width.

tsuoanttila committed:
Enable long tap emulation on iOS for context clicks (#8565) (#8570)

Fix release notes mention of context click events

alejandro-du committed:
Updated links to zip files in tutorial (#8659)

* Updated links to zip files in tutorial

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vaadin/framework?pullRequest=8720) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/vaadin/framework?pullRequest=8720) before we can accept your contribution.<br/>**16** out of **20** committers have signed the CLA.<br/><br/>:white_check_mark: hesara<br/>:white_check_mark: Legioth<br/>:white_check_mark: mstahv<br/>:white_check_mark: elmot<br/>:white_check_mark: wbadam<br/>:white_check_mark: alejandro-du<br/>:white_check_mark: Artur-<br/>:white_check_mark: thinwire<br/>:white_check_mark: snyholm<br/>:white_check_mark: johannest<br/>:white_check_mark: tsuoanttila<br/>:white_check_mark: pleku<br/>:white_check_mark: denis-anisimov<br/>:white_check_mark: OlliTietavainenVaadin<br/>:white_check_mark: caalador<br/>:white_check_mark: ahie<br/>:x: ingokegel<br/>:x: emarc<br/>:x: Flamenco<br/>:x: Johannes Dahlström<br/><hr/>**Johannes Dahlström** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vaadin/framework?pullRequest=8720) it.</sub>

tsuoanttila committed:
Remove unnecessary full classnames and parenthesis from code (#8681)

This change aims to reduce the number of false positives when comparing Vaadin 7 compatibility package in Vaadin 8 and actual Vaadin 7.7 branch. Conflicting parts are moved to imports as much as possible. Files have been formatted with same Eclipse version.

pleku commented:
Hello, I don't know why this PR has been made, but it makes absolutely no sense.

>One Does Not Simply
>
>Merge 7.7 changes to 8

We have picked and are currently in the progress of picking the last changes from 7.7 to 8.0.
